image/image_decoder.h

```diff
@@ -113,6 +113,22 @@ class ImageDecoder {
 	virtual bool hasTransparentColor() const { return false; }
 	/** Return the transparent color. */
 	virtual uint32 getTransparentColor() const { return 0; }
+
+	/**
+	 * Get the mask data for the decoded image.
+	 */
+	virtual const byte *getMask() const { return 0; }
```

Comment:
Would it be possible for this to return a CLUT8 `Graphics::Surface` to match our common blitting code?

Reply:
> Would it be possible for this to return a CLUT8 `Graphics::Surface` to match our common blitting code?

Sure. I wasn't aware of this before. Is it the way you meant now?

Reply:
It looks OK, although `hasMask()`, `getMaskRowBytes()` and `getMaskHeight()` in the base `ImageDecoder` class shouldn't be needed any more.

As an aside, it should be possible to make use of the `maskBlitFrom` function from PR #6377 to simplify the `drawSprite` function now that the standard format is being used.

Reply:
> It looks OK, although `hasMask()`, `getMaskRowBytes()` and `getMaskHeight()` in the base `ImageDecoder` class shouldn't be needed any more.

Oops, I missed those in the rush. Thanks! Though `hasMask()` seems analogous to `hasPalette()` so I'm leaving that one for now.